### PR TITLE
NTSV Tegus-class Logistics Freighter

### PR DESCRIPTION
**Pull request by goober3**
## About The Pull Request
image

ough
Adds a Nanotrasen Logistics Freighter with a large shuttle for performing mining & engineering operations. 

A full telecomms setup, a relay linking the shuttle with the mothership, and a spyglass kit facilities Field Technician to Mothership communication, allowing expedition teams to travel long distances whilst still remaining connected to the Tegus.

This PR also ports Fikou's spyglass rework from [here](https://github.com/tgstation/tgstation/pull/53355) (this was included in the original Columba PR, which is closed now!)
and the Tegu from [here](https://github.com/vlggms/tegustation/pull/97)
description of the Tegu from Valorium

## Why It's Good For The Game

Columba But Better
shiptest's most unatomized ship PR 2023

## Changelog

:cl:
add: The Tegus-class Logistics Freighter, a Nanotrasen mining / engineering ship with a large subshuttle.
add: The Tegu, sprite by Aspev 
tweak: Spyglasses use actions now
add: The Micro SMES, for use on subshuttles
add: The Telecomms Parts design disk, prints telecomms parts from an autolathe.
/:cl:

### CODE REVIEW

**Review by rye-rice — APPROVED** [review]
i fucking love tegus you jsut dont understand

**Review by spockye — CHANGES_REQUESTED** [review]
image

unconnected vent and scrupper


image

no need for the manifold to go into the wall

image

please replace it with "tegus_shield_port"! 

the ship looks rad. Good work!

**Comment by goober3** [comment]
the checks have somehow fixed themselves. Good.

**Comment by goober3** [comment]
the checks have somehow unfixed themselves. Bad.

File: code/datums/shuttles.dm [hunk 1]
```diff
@@ -351,5 +351,9 @@
 /datum/map_template/shuttle/subshuttles
 	category = "subshuttles"
 
-//your subshuttle here
+/datum/map_template/shuttle/subshuttles/nanotrasen_tegus
+	file_name = "nanotrasen_shuttle_tegus"
+	name = "Tegus Transport Shuttle"
```
Comment (Erikafox) [inline code/datums/shuttles.dm]: Please give it a different name than the ship

**Review by Erikafox — CHANGES_REQUESTED** [review]
image
The shutters on the subshuttle need to start closed. You get weird space wind and trouble with freespawning it otherwise.
I'd also like it one of the holofields was replaced with a cycling airlock. 
StrongDMM_h1SqbcNbbB
lockers go on 28/-28
StrongDMM_q3PToBHOOH
There needs to be a cable under the terminal for it to work
image
Use /three_quarters to reduce file size. 
StrongDMM_XKF1HRWOJq
Give this room more of a tech bend. Try having some techfloor under the machines. To my eye it blends together with the rest of the ship far too much
StrongDMM_m7M2YXw4aq
Ditto down here
image
Where is THEIR BED
Also lockers
image
Don't mix catwalks and pumps. It looks like shit.
StrongDMM_LQ3tRvCGkF
Set the density on the booze-o to 0 and pixelshift it onto a wall. 
StrongDMM_ywAI8RXSNy
Firelocks on stairs can ruin the vertical effect. Consider moving it back
StrongDMM_Qw3nDCki29
Run the supply pipe with the scrubber pipe so you don't have atmos machine - pipe overlap
image
Either or. If you want them to have the choice give them the boards and make it so they have to construct them.
StrongDMM_k8bs1svMGD
MAKE IT SYMMETRICAL 
StrongDMM_GK3iNgy5Y7
I'd have airlocks leading into this room. Also ditch the stairs and the catwalks. Make it seem more like an office - ready room. 


Right now the entire ship feels very _flat_ I think this is because you only used 3-4 types of tiles, some stairs, and decalling. Flat isn't bad, but it can make things less interesting. I'd experiment with stronger decal colors, different tile types, and establishing theming to areas, because some of the placement feels flat out random to me. The mining room and bar are probably the most stand out offenders to this. 
StrongDMM_P9ZuzjL3SB
Also consider using some interior reinforced hull tiles here.

image
Honestly the nose could be expanded to give it some proper cargo areas. It really feels like a shape first design second ship, leading to things like the semi-random filing cabinets, or the mining room.
StrongDMM_ce88xk58lV
Airlocks this size should have 2 vents for quick cycling.

Consider making a subtype of Ions that are less efficient to encourage usage of the subshuttle. You can't really force it but it should be the preferable alternative.

**Comment by goober3** [comment]
review jumpscare

**Review by Erikafox — CHANGES_REQUESTED** [review]
Needs a docking point for subshuttles on the outside.

**Review by spockye — COMMENTED** [review]
image

missing pipes

image

the canteen feels really secluded. mabye add some windows here?

image

give the poor dude a bed :(


image

this apc isnt wired


-----------

mabye add a locker with spare clothes? not seeing any

**Review by Latentish — DISMISSED** [review]
To start you have two atmos piping errors and a subshuttle related runtime that need to be fixed before this can be merged.
This airlock on the subshuttle is not set up properly
image

Cryopod wakeups should be separate from crew quarters, or at least have some more separation.
image

You need to add lightswitches

Not mechanically supported, but I definitely prefer telecomms be stored in a vacuum or cold temperature.
image

You need mroe emergency lockers.
image


Cryopod areas also need handheld radios and lights somewhere reachable.
Also this are looks more like a front desk office than crewquarters.
image

The subshuttle could definitely use a pair of emergency lockers.

**Comment by Latentish** [comment]
Ship's pretty good overall.

**Comment by goober3** [comment]
this ready for TM / review, aside from the funky integration check fail. i'm not quite sure how to fix that

**Review by Erikafox — COMMENTED** [review]
image
I think this room needs a lil more decalling

Maybe go a bit heavier on the metal interior, instead of titanium.

**Comment by PrefabQuasar** [comment]
already thoroughly reviewed, so i wont pile on much, but it's worth considering reducing the range of that subshuttle by cutting out the pacman, and maybe the SMES too, otherwise it really can just go Wherever

**Review by spockye — APPROVED** [review]
Massive improvements all around. I'm liking where this is going!

**Comment by Latentish** [comment]
I have to say that I really preferred the overall shape of your last iteration more.

**Comment by goober3** [comment]
> I have to say that I really preferred the overall shape of your last iteration more.

perhaps. i think this shape still works aesthetically. and the layout has improved moreso, i hope


> image I think this room needs a lil more decalling
> 
> Maybe go a bit heavier on the metal interior, instead of titanium.

i'll Experiment

**Comment by goober3** [comment]
OUGH

**Review by Apogee-dev — CHANGES_REQUESTED** [review]
Lock the subshuttle cockpit's airlock to the pilot job, please. It makes sense to keep the other crew members from running off with it, and if somebody desperately needs in they can always hack the door.

**Comment by Erikafox** [comment]
No cryopod computer in the cryopod room

**Comment by goober3** [comment]
> No cryopod computer in the cryopod room

this is being fixed!

**Review by Erikafox — CHANGES_REQUESTED** [review]
descending from the heavens to inform you that I like the ship, except engineering
The atmospherics machinery needs to be better marked, and it could use an atmospheric siphon set up (as well as air pressure tank, but that's less essential)

**Comment by goober3** [comment]
Ough

**Review by Erikafox — APPROVED** [review]
it'll probably be fine

**Review by Latentish — DISMISSED** [review]
Pass your checks.

**Review by Apogee-dev — APPROVED** [review]

**Comment by Erikafox** [comment]
waiting for the harddel fixes to go in because that fixes the CL failure this has.

**Comment by goober3** [comment]
oh no way!!! cool!!!

**Review by spockye — CHANGES_REQUESTED** [review]
image

Poster light

also add lightswitches

**Comment by Zevotech** [comment]
## Please for the love of god add lightswitches to the ship and subshuttle
Replace these with the new good rations
image

Give the other plasma pacman 5 sheets as well, or maybe replace it with some other kind of pacman for diversity. A bad roll on the solars might cause power issues. Also see if you can put the fuel in a crate or something. Having it sat on top of the generator looks weird.
image

Gngineering in general looks kind of empty, maybe add a locker or a desk for whoever works back here, a power monitoring console would also be appreciated. Also, replace the air canister with a high pressure airtank. You'll get maybe five uses out of the airlocks before the canister runs out and they stop working. The airlocks also straight up dont work until the canister is wrenched down.
image

Give the empty crates some junk or spare mining gear maybe. Add more trash around, maybe a crate shelf, make it FEEL like a cargo bay. Also, probably move that pipe dispenser to engineering. Not sure why its just sitting there.
image

Did you copy this disk from the vela/cepheus? see if you can get rid of those CMM designs on there.
image

Feels like there could be more in here. Maybe toss in a few cleaner grenades, a maid outfit or some maintenance loot. 
image

Maybe make the mining camera a small item so it fits into pockets. Also the captains locker has independent captains gloves in it. I like the emergency signaler, nice touch. You can change the ships starting funds in the JSON, maybe give it a bit more than 2000 considering its dedicated to logistics?
image

Throw a crate at the end for the refined materials to be put into. Otherwise, very nice smelting setup! Functions well.
image

This computer doesn't seem to work. Also, why is it here? I'd replace it with something else.
image

I'd recommend throwing on some exterior defenses similar to the ones on the Vela to block potential debris or meteors. Good way to use this empty space.
image

1. The subshuttle starts with no fuel. 2. The prechargers are from the fucking PILL CLASS. they have 0 DENSITY. Please use normal prechargers. 3. The airlock on the subshuttle does not work. Might be related to how the layer adapters are placed, look at other ships airlocks to see how they do it. 4. Either add access buttons to the outside of the subshuttle or move them to where they can be pressed from both inside and outside. 5. Give the pilot a proximity crew pinpointer instead of a normal one.
image
image

**Comment by goober3** [comment]
yeah. i think i'm going to shelve this for a remap in the future. this ship could be much, much better.

**Pull request closed without merging**